### PR TITLE
fix(QTM-315): Adjusts component to receive a text element and an action

### PR DESCRIPTION
**Pull request by toncabral88**
## Description
https://jirasoftware.catho.com.br/browse/QTM-315

https://www.figma.com/file/SED32ByxEn8C1HR5J5aR2JGs/Style-Guide---Design-System?node-id=150%3A0

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] TypeScript updated

### A11y review
- [ ] A11y checks
  - [ ] accessible via keyboard?
  - [ ] Identifiable through assistive technology (screen-reader chrome extension)?
  - [ ] Semantically correct html elements (or if necessary identified by WAI-ARIA)?
  - [ ] Is there a deficiency in color contrast?

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

### CODE REVIEW

**Review by matheusrpp — DISMISSED** [review]
I leaving one a11y option unchecked. I tought it was weird how the screen reader was reading the snackbar when you navigate to the container. I pointed out to Cael, because I don't know if it's supposed to be like that or not, so I'm leaving for him to decide.



## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review
- [x] TypeScript updated

### A11y review
- [ ] A11y checks
  - [x] accessible via keyboard?
  - [ ] Identifiable through assistive technology (screen-reader chrome extension)?
  - [x] Semantically correct html elements (or if necessary identified by WAI-ARIA)?
  - [ ] Is there a deficiency in color contrast?

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile

File: components/SnackBar/index.d.ts [hunk 1]
```diff
@@ -16,11 +16,12 @@ export interface SnackBarProps {
     };
   };
   closeButtonAriaLabel?: string;
-  onClose?: React.MouseEventHandler<HTMLButtonElement>;
+  onClose?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement> | React.TouchEventHandler<HTMLButtonElement | HTMLAnchorElement>;
   secondsToClose?: number;
   skin?: 'primary' | 'success' | 'error' | 'neutral' | 'warning';
-  text?: string;
+  text: string | HTMLParagraphElement;
```
Comment (rapahaeru) [inline components/SnackBar/index.d.ts:22]: shouldnt be `text?`

File: components/SnackBar/SnackBar.jsx [hunk 2]
```diff
@@ -114,47 +125,65 @@ const SkinIcon = styled(Icon)`
...
-  align-items: center;
-`;
+    return css`
+      background: ${hexToRgba(color, 0.15)};
 
-const ActionButton = styled(Button)``;
+      margin-left: ${small}px;
+      box-shadow: none;
+      color: ${color};
+      outline: none;
```
Comment (rapahaeru) [inline components/SnackBar/SnackBar.jsx:170]: i think of removes the outline affects the a11y.
how about of keep it?

**Review by rapahaeru — APPROVED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] TypeScript updated

### A11y review
- [x] A11y checks
  - [x] accessible via keyboard?
  - [x] Identifiable through assistive technology (screen-reader chrome extension)?
  - [x] Semantically correct html elements (or if necessary identified by WAI-ARIA)?
  - [x] Is there a deficiency in color contrast?

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile
- [x] Ux review validation

File: components/SnackBar/index.d.ts (hunk 1 above)
Comment (toncabral88) [reply components/SnackBar/index.d.ts:22]: Isn't text mandatory?

Comment (rapahaeru) [reply components/SnackBar/index.d.ts:22]: You're right!

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**